pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -213,88 +214,93 @@ var _ = g.Describe("[Feature:Machines] Autoscaler should", func() {
-
-		g.By("Creating MachineAutoscaler objects")
-		machineAutoscaler := machineAutoscalerResource(&targetMachineSet, 1, initialNumberOfReplicas+1)
-		err = client.Create(context.TODO(), machineAutoscaler)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("Labeling all nodes belonging to the machineset (before scale up phase)")
-		err = labelMachineSetNodes(client, &targetMachineSet, nodeTestLabel)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		glog.Info("Get nodeList")
-		nodeList := corev1.NodeList{}
-		err = client.List(context.TODO(), runtimeclient.MatchingLabels(map[string]string{nodeTestLabel: ""}), &nodeList)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		nodeGroupInitialTotalNodes := len(nodeList.Items)
-		glog.Infof("Cluster initial number of nodes in node group %v is %d", targetMachineSet.Name, nodeGroupInitialTotalNodes)
+		initialNumberOfReplicas0 := pointer.Int32PtrDerefOr(targetMachineSet0.Spec.Replicas, e2e.DefaultMachineSetReplicas)
+		initialNumberOfReplicas1 := pointer.Int32PtrDerefOr(targetMachineSet1.Spec.Replicas, e2e.DefaultMachineSetReplicas)
+		glog.Infof("initialNumberOfReplicas0 %d, initialNumberOfReplicas1 %d", initialNumberOfReplicas0, initialNumberOfReplicas1)
+		o.Expect(initialNumberOfReplicas0).To(o.BeNumerically("==", initialNumberOfReplicas1))
 
 		g.By("Creating workload")
 		err = client.Create(context.TODO(), newWorkLoad())
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		g.By("Waiting for cluster to scale out number of replicas")
-		err = wait.PollImmediate(5*time.Second, e2e.WaitLong, func() (bool, error) {
-			ms, err := e2e.GetMachineSet(context.TODO(), client, targetMachineSet.Name)
-			if err != nil {
-				glog.Errorf("error getting machineset object: %v, retrying...", err)
-				return false, nil
-			}
-			glog.Infof("MachineSet %s. Initial number of replicas: %d. Current number of replicas: %d", targetMachineSet.Name, initialNumberOfReplicas, pointer.Int32PtrDerefOr(ms.Spec.Replicas, e2e.DefaultMachineSetReplicas))
-			return pointer.Int32PtrDerefOr(ms.Spec.Replicas, e2e.DefaultMachineSetReplicas) > initialNumberOfReplicas, nil
-		})
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("Waiting for cluster to scale up nodes")
-		err = wait.PollImmediate(5*time.Second, e2e.WaitLong, func() (bool, error) {
-			scaledMachines := mapiv1beta1.MachineList{}
-			if err := client.List(context.TODO(), runtimeclient.MatchingLabels(targetMachineSet.Spec.Selector.MatchLabels), &scaledMachines); err != nil {
-				glog.Errorf("Error querying api for machineset object: %v, retrying...", err)
-				return false, nil
-			}
-
-			// get all linked nodes and label them
-			nodeCounter := 0
-			for _, machine := range scaledMachines.Items {
-				if machine.Status.NodeRef == nil {
-					glog.Errorf("Machine %q does not have node reference set", machine.Name)
+		scaleUpFunc := func(targetMachineSet mapiv1beta1.MachineSet, nodeTestLabel string, initialNumberOfReplicas int32, expectedReplicas int32) {
```

Comment:
Consider moving this to the beginning of the function as it can help identify unintended variable shadowing. Ditto for the `scaleDown` func.